### PR TITLE
315 implement admin panel

### PR DESCRIPTION
**Pull request by LucasTesnier**
Implementation of the AdminPanel !

Link the API library inside the Server and Client executable :
- Create an instance of the class
- Update the database creation process by adding a new player named admin with admin password and moderator right's
- Make the call to the API when necessary (Auth, Join -> Banned, Muted -> Chat, GamePlayed -> Room Start, KilledEnemies -> Enemy death, Deaths -> Player death)

Implement a new 4X protocol inside Communicator and Server :
- Add get request for all the database value
- Add set request for Moderator, Mute, Ban, Pseudo, Password
- Use the Muted value when message is sended.

Implement the Admin Panel :
- Create a new executable
- Get current command and parse it
- Create inside the Notion an Admin Panel documentation
- Create a get command for specified data or all
- Create some Moderator command
- Create some Administrator command
- Implement an auth system before the start of Admin panel. (Your player need to have the moderator rights. By default only the user admin admin could have it)

Prepare the AdminPanel for the Pseudo and Password size refacto.

Fix the Room creation segfault.

### CODE REVIEW

**Review by osvegn — APPROVED** [review]

File: Room/main.cpp [hunk 1]
```diff
@@ -13,11 +13,14 @@ using namespace argument_handler;
 
 int main(int ac, char **av)
 {
+    std::cerr << "HEY POTO" << std::endl;
```
Comment (31Nathan) [inline Room/main.cpp:16]: BAH EH OH

File: Room/main.cpp [hunk 2]
```diff
@@ -13,11 +13,14 @@ using namespace argument_handler;
...
     ArgumentHandler argumentHandler = ArgumentHandler(ac, av);
     ArgumentHandler::RoomInformation roomInformations = argumentHandler.extractRoomInformations();
 
+    std::cerr << "???" << std::endl;
```
Comment (31Nathan) [inline Room/main.cpp:20]: MY DEAR FRIEND

File: Server/RoomInstance.cpp [hunk 3]
```diff
@@ -26,10 +26,11 @@ namespace server_data
         _terminated = false;
         _input = new boost::process::opstream();
         _output = new boost::process::ipstream();
+        std::cerr << "LOL" << std::endl;
```
Comment (31Nathan) [inline Server/RoomInstance.cpp:29]: BON MON REUF

File: Server/Server.cpp [hunk 4]
```diff
@@ -119,7 +162,10 @@ void Server::_holdACreateRoomRequest(CommunicatorMessage createDemand)
         }
     }
 
+    std::cerr << "C" << std::endl;
```
Comment (31Nathan) [inline Server/Server.cpp:165]: ANOTHER ONE

File: Room/main.cpp (hunk 1 above)
Comment (osvegn) [inline Room/main.cpp:16]: INVALID

**Review by osvegn — DISMISSED** [review]

File: Room/main.cpp (hunk 1 above)
Comment (TomDUVAL-MAHE) [inline Room/main.cpp:16]: debug

File: Room/main.cpp (hunk 2 above)
Comment (TomDUVAL-MAHE) [inline Room/main.cpp:20]: debug

**Review by TomDUVAL-MAHE — CHANGES_REQUESTED** [review]

File: libs/R-TypeLogic/Server/Systems/CollidableSystem.cpp [hunk 5]
```diff
@@ -69,25 +85,50 @@ void Collide::collide(
                         sndLife.lifePoint = fstDamage.damagePoint;
                     sndLife.lifePoint -= fstDamage.damagePoint;
                     sndLife.modified = true;
+                    if (sndLife.lifePoint == 0 && fstEntity->contains<Player>() && sndEntity->contains<Enemy>()) {
```
Comment (31Nathan) [inline libs/R-TypeLogic/Server/Systems/CollidableSystem.cpp:88]: This functionality does not have anything to do here, it should be in death system or something similar

**Review by 31Nathan — CHANGES_REQUESTED** [review]

File: Room/main.cpp (hunk 1 above)
Comment (LucasTesnier) [reply Room/main.cpp:16]: :eyes:

Comment (LucasTesnier) [reply Room/main.cpp:16]: :eyes:

File: Room/main.cpp (hunk 2 above)
Comment (LucasTesnier) [reply Room/main.cpp:20]: :eyes:

File: Room/main.cpp (hunk 1 above)
Comment (LucasTesnier) [reply Room/main.cpp:16]: :eyes:

File: Room/main.cpp (hunk 2 above)
Comment (LucasTesnier) [reply Room/main.cpp:20]: :eyes:

File: Server/RoomInstance.cpp (hunk 3 above)
Comment (LucasTesnier) [reply Server/RoomInstance.cpp:29]: :eyes:

File: Server/Server.cpp (hunk 4 above)
Comment (LucasTesnier) [reply Server/Server.cpp:165]: :eyes:

File: libs/R-TypeLogic/Server/Systems/CollidableSystem.cpp (hunk 5 above)
Comment (LucasTesnier) [reply libs/R-TypeLogic/Server/Systems/CollidableSystem.cpp]: :eyes:

**Review by 31Nathan — DISMISSED** [review]

**Review by TomDUVAL-MAHE — APPROVED** [review]

**Review by TomDUVAL-MAHE — APPROVED** [review]

**Review by 31Nathan — APPROVED** [review]

**Pull request merged**